### PR TITLE
[FLINK-3486] [tableAPI] Fix broken renaming of all fields.

### PR DESCRIPTION
**Pull request by fhueske**

### CODE REVIEW

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/table.scala [hunk 1]
```diff
@@ -98,7 +100,19 @@ class Table(
       .map(toRexNode(_, relBuilder))
 
     relBuilder.project(exprs.toIterable.asJava)
-    new Table(relBuilder.build(), relBuilder)
+    var projected = relBuilder.build()
+
+    if(relNode == projected) {
+      // Calcite's RelBuilder does not translate identity projects even if they rename fields.
+      //   Add a projection ourselves (will be automatically removed by translation rules).
+      val names = exprs.map(_.asInstanceOf[RexCall].getOperands.get(1)
```
Comment (vasia) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/table.scala]: This cast to `RexCall` seems to cause several tests to fail.

Comment (fhueske) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/table.scala]: Woops! 
Thanks, I'll check that.

**Comment by fhueske** [comment]
I fixed the failing tests.

**Comment by fhueske** [comment]
Will merge this to `tableOnCalcite` branch.

**Comment by fhueske** [comment]
Merged to `tableOnCalcite`

**Pull request closed without merging**